### PR TITLE
27 get favorite

### PR DESCRIPTION
**Pull request by MohammedAlghazali**
based on #27 i made
* route /favorite/:userId to get all favorite courses for specific user
* make function getFavorite for this route
* make query getFavoriteCourse, getUserById needed for this function
* insert fake data to make tests
* make test for the route and for the both query

### make request by postman

---

### CODE REVIEW

File: server/src/controllers/middleware/getFavorite.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+          res.json({
+            message: 'There is no favorite courses for this user id',
+          });
+        }
+      } else {
+        res.status(404).json({
+          message: 'Invalid id',
+        });
+      }
+    }
```
Comment (omar-dahdouh) [inline server/src/controllers/middleware/getFavorite.js:22]: if `userId` is less than or equal to zero no response will return
remove the check `if (userId > 0) { ... } `
OR add `else` case
```js
if (userId > 0) {
    ...
} else {
    res.status(400).json({
        message: 'invalid user id',
    });
}
```

Comment (MohammedAlghazali) [reply server/src/controllers/middleware/getFavorite.js:22]: yes, thank you I forget to make a schema for this params.

File: server/src/controllers/middleware/getFavorite.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  } catch (err) {
+    if (err.errors) {
+      res.status(404).json({
+        message: err.errors,
+      });
```
Comment (AlaaTaima) [inline server/src/controllers/middleware/getFavorite.js:29]: **It's a good way to handle validation error but, there are some points needed to be refactored** 
here, you want to handle validation error of `userId` so, the status code should be 400 (bad request) and this syntax will be more clear : 
```suggestion
        if (err.name === 'ValidationError') {
      return res
        .status(400)
        .json({ message: 'invalid inputs' });
    }
      

```

File: server/src/database/config/fakeData.sql [hunk 3]
```diff
@@ -0,0 +1,44 @@
+INSERT INTO users (name , email) VALUES ('Mohammed', 'hamood-monzer11@hotmail.com'),  ('omar', 'omar@hotmail.com');
+
+INSERT INTO course(category_id,title,author_name,url,rate,reviews,description, source)
+VALUES(
+    '1',
+    'computer',
+    'Moh',
+    'www.course1.com',
```
Comment (AlaaTaima) [inline server/src/database/config/fakeData.sql:8]: **here, the course url is edited to be unique so, fix your fake data**

File: server/src/controllers/middleware/getFavorite.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+          res.json(rows);
+        } else {
+          res.json({
+            message: 'There is no favorite courses for this user id',
+          });
+        }
+      } else {
+        res.status(404).json({
+          message: 'Invalid id',
+        });
```
Comment (AlaaTaima) [inline server/src/controllers/middleware/getFavorite.js:22]: **Really, you did a good job in handling all cases here :+1:**  but, there is a small point : 
* `if (rows[0]) ` , here you want to check if the rows is empty or not so, you can use `rows.length `instead

Comment (MohammedAlghazali) [reply server/src/controllers/middleware/getFavorite.js:22]: Thank you,
But here will not be any difference in the two case

Comment (MohammedAlghazali) [reply server/src/controllers/middleware/getFavorite.js:22]: If row is empty row[0] will be undefined so the condition will not pass

Comment (AlaaTaima) [reply server/src/controllers/middleware/getFavorite.js:22]: yes, no difference but at the first shot I think you checked for rows[0] not for the length

**Pull request merged**